Updated Documentation for Manual Setup (#889)
ubccr/xdmod: docs/databases.md

title: Database Guide
---

Open XDMoD uses several MySQL databases.  These will be
automatically created by the database section of the `xdmod-setup`
command.

Manual Setup
------------

**NOTE**: Manual setup instructions are provided for informational purposes only; no guarantee of support is provided.

If you prefer to not give the root (or any other privileged) user's

> credentials to the setup command, you will need to create and

Reviewer: ```suggestion
credentials to the `xdmod-setup` command, you will need to create and
```

    mysql> GRANT ALL ON modw_filters.*    TO 'username'@'localhost' IDENTIFIED BY 'password';

You must also run the following SQL:

    mysql> GRANT TRIGGER, DROP, INDEX, CREATE, INSERT,
    SELECT, DELETE, UPDATE, CREATE VIEW, SHOW VIEW,
    ALTER, SHOW DATABASES, CREATE TEMPORARY TABLES,
    CREATE ROUTINE, ALTER ROUTINE, EVENT, RELOAD, FILE,
    CREATE TABLESPACE, PROCESS, REFERENCES,
    LOCK TABLES ON *.* TO 'username'@'localhost';

You will need to change `localhost` if you are running MySQL on a
different machine than where you installed Open XDMoD.

Initialize databases:

    $ mysql -u username -ppassword mod_logger      </usr/share/xdmod/db/schema/mod_logger.sql
    $ mysql -u username -ppassword mod_logger      </usr/share/xdmod/db/data/mod_logger.sql

Bootstrap and populate tables:

    $ /usr/share/xdmod/tools/etl/etl_overseer.php -p xdmod.shredder-bootstrap
    $ /usr/share/xdmod/tools/etl/etl_overseer.php -p xdmod.xdb-bootstrap
    $ /usr/share/xdmod/tools/etl/etl_overseer.php -p xdmod.hpcdb-bootstrap
    $ /usr/share/xdmod/tools/etl/etl_overseer.php -p xdmod.staging-ingest-common

>     $ /usr/share/xdmod/tools/etl/etl_overseer.php -p xdmod.jobs-xdw-bootstrap

Reviewer: Since you refer to a source code install below, then you perhaps should do the same here since the full path will be different for a source code install.

Manual Setup
------------

**NOTE**: Manual setup instructions are provided for informational purposes only; no guarantee of support is provided.

If you prefer to not give the root (or any other privileged) user's
credentials to the setup command, you will need to create and
initialize the databases manually and add the database credentials to
`portal_settings.ini`. These databases must be named as shown below.
Also, the credentials for `modw`, `modw_aggregates`, and `modw_filters`
must be the same.

**NOTE**: `modw_aggregates` and `modw_filters` aren't listed in
`portal_settings.ini`.

Create databases and database user:

>     mysql> CREATE DATABASE mod_hpcdb;

Reviewer: CREATE DATABASE modw_cloud;

Manual Setup
------------

> 

Reviewer 2: Was my grammar incorrect here?  The intent was "user's username and password", not multiple users.

> **NOTE**: Manual setup instructions are provided for informational purposes only; no guarantee of support can be provided.

Reviewer 2: ```suggestion
**NOTE**: Manual setup instructions are provided for informational purposes only; no guarantee of support is provided.
```

    mysql> GRANT ALL ON modw_aggregates.* TO 'username'@'localhost' IDENTIFIED BY 'password';
    mysql> GRANT ALL ON modw_filters.*    TO 'username'@'localhost' IDENTIFIED BY 'password';

> You should also run the following SQL:

Reviewer 3: format this like code directly above since this should also be run on the mysql prompt

> You should also run the following SQL:

Reviewer 2: Should or must?

>     $ mysql -u username -ppassword modw_aggregates </usr/share/xdmod/db/schema/modw_aggregates.sql

Reviewer 3: you removed the section about the schema being located in `/usr/share/xdmod/db/schema` but left this section that references it, why?

Author: Actually not sure we want to keep this section at all since we're explicitly running all these pipeline commands, was hoping for other people's feedback on this.

Reviewer 2: Check which files still contain SQL.  Some may still be in use.

    $ mysql -u username -ppassword modw_filters    </usr/share/xdmod/db/schema/modw_filters.sql

Bootstrap and populate tables:

>     $ php XDMOD_PATH/tools/etl/etl_overseer.php -p xdmod.shredder-bootstrap

Reviewer 3: `etl_overseer.php` is executable and does not the need the `php` in front of it

use `/usr/share/xdmod` instead of `XDMOD_PATH` to be consistent with the rest of the document

Initialize ACLs:

    $ acl-config

**NOTE**: If you installed the source tarball update your `PATH` to include the

> Open XDMoD `bin` directory before running these commands.

Reviewer: s/these/this/

Reviewer: s/commands/command/
